templates/app/views/carts/_cart_item.html.erb

```diff
@@ -26,8 +26,8 @@
 
     <div class="flex flex-row-reverse justify-between col-span-full sm:col-span-4 sm:flex-col">
       <div class="flex items-center justify-between w-full">
-        <div class="cart-item__quantity select-input [&>select]:min-w-[100px]">
-          <%= item_form.select :quantity, (0..variant.total_on_hand).to_a %>
+        <div class="cart-item__quantity number-input mr-4">
+          <%= item_form.number_field :quantity, class: "rounded-md border border border-gray-mid rounded px-4 py-3 placeholder:text-gray-500 text-black text-body w-full hover:border-gray-400 focus:outline-none focus:ring focus:ring-offset-1 focus:ring-black/20 focus:border-gray-400 dark:bg-gray-800 dark:border-gray-600 dark:text-sand dark:placeholder:text-gray-400 dark:focus:ring-white/30" %>
```

Comment:
These are the tailwind classes from the coupon code input. I personally think it's a bit much, but hey.

Reply:
We could a `max` on the number field and set this to `variant.total_on_hand`

Reply:
what if `variant.total_on_hand` is Infinity? Can the `max` property deal with that?

Reply:
Not even sure what `Infinity.to_s` is 🤣